src/experiments.js

```diff
@@ -56,6 +64,126 @@ export function isCanary(win) {
 
 
 /**
+ * Enable experiments detailed in an origin trials token iff the token is
+ * valid. A token is invalid if *
+ *   1. The token is malformed (e.g. non-existant version number)
+ *   2. The token is not for this origin
+ *   3. The experiments data was not signed with our private key
+ * @param {!Window} win
+ * @param {string} token
+ * @param {!Object=} opt_publicJwk Used for testing only.
+ * @return {!Promise}
+ */
+function enableExperimentsFromToken(win, token, opt_publicJwk) {
+  const crypto = cryptoFor(win);
+  if (!crypto.isCryptoAvailable()) {
+    user().warn(TAG, 'Crypto is unavailable');
+    return Promise.resolve();
+  }
+  /**
+   * token = encode64(version + length + config + sign(config, private_key))
+   * version = 1 byte version of the token format (starting at 0x0)
+   */
+  let current = 0;
+  const bytes = stringToBytes(atob(token));
+  const version = bytes[current];
+  if (version !== 0) {
+    // Unrecognized version number
+    return Promise.reject(`Unrecognized experiments token version: ${version}`);
+  }
+  current++;
+  /**
+   * Version 0:
+   * length = 4 bytes representing number of bytes in config
+   * config = string containing the experiment ID, origin URL, etc.
+   */
+  const bytesForConfigSize = 4;
+  const configSize =
+      bytesToUInt32(bytes.subarray(current, current + bytesForConfigSize));
+  current += bytesForConfigSize;
+  if (configSize > bytes.length - current) {
+    return Promise.reject('Specified len extends past end of buffer');
+  }
+  const configBytes = bytes.subarray(current, current + configSize);
+  current += configSize;
+  const signatureBytes = bytes.subarray(current);
+
+  // TODO(kmh287, choumx) fill in real public key
```

Comment:
This may need to be filled in later once we determine the proper way to store the key pair.